model/atmosphere/subgrid_scale_physics/muphys/tests/graupel/test_graupel.py

```diff
@@ -0,0 +1,408 @@
+# ICON4Py - ICON inspired code in Python and GT4Py
+#
+# Copyright (c) 2022-2024, ETH Zurich and MeteoSwiss
+# All rights reserved.
+#
+# Please, refer to the LICENSE file in the root directory.
+# SPDX-License-Identifier: BSD-3-Clause
+import functools
+
+import numpy as np
+import pytest
+
+from icon4py.model.common import dimension as dims
+from icon4py.model.common.decomposition import definitions
+from icon4py.model.common.grid import (
+    geometry as geometry,
+    geometry_attributes as attrs,
+    horizontal as h_grid,
+    icon,
+    simple as simple,
+)
+from icon4py.model.common.grid.geometry import as_sparse_field
+from icon4py.model.common.test_utils import datatest_utils as dt_utils, helpers
+from icon4py.model.common.utils import gt4py_field_allocation as alloc
+
+from . import utils
+
+
+grid_geometries = {}
+
+
+def get_grid_geometry(backend, grid_file) -> geometry.GridGeometry:
+    def construct_decomposition_info(grid: icon.IconGrid) -> definitions.DecompositionInfo:
+        edge_indices = alloc.allocate_indices(dims.EdgeDim, grid)
+        owner_mask = np.ones((grid.num_edges,), dtype=bool)
+        decomposition_info = definitions.DecompositionInfo(klevels=grid.num_levels)
+        decomposition_info.with_dimension(dims.EdgeDim, edge_indices.ndarray, owner_mask)
+        return decomposition_info
+
+    def construct_grid_geometry(grid_file: str):
+        gm = utils.run_grid_manager(grid_file)
+        grid = gm.grid
+        decomposition_info = construct_decomposition_info(grid)
+        geometry_source = geometry.GridGeometry(
+            grid, decomposition_info, backend, gm.coordinates, gm.geometry, attrs.attrs
+        )
+        return geometry_source
+
+    if not grid_geometries.get(grid_file):
+        grid_geometries[grid_file] = construct_grid_geometry(grid_file)
+    return grid_geometries[grid_file]
+
+
+def test_geometry_raises_for_unknown_field(backend):
+    geometry = get_grid_geometry(backend, dt_utils.R02B04_GLOBAL)
+    with pytest.raises(ValueError) as e:
+        geometry.get("foo")
+        assert "'foo'" in e.value
+        assert "'GridGeometry'" in e.value
```

Comment:
should both of these assertions fail?

Reply:
This is the python test program from the muphys-cpp version when using Python bindings.  It serves as the guide to implement python_test.py and should probably be removed now.